authors/apps/authentication/serializers.py

```diff
@@ -4,26 +4,51 @@
 
 from .models import User
 
+import re
+
 
 class RegistrationSerializer(serializers.ModelSerializer):
     """Serializers registration requests and creates a new user."""
 
     # Ensure passwords are at least 8 characters long, no longer than 128
     # characters, and can not be read by the client.
-    password = serializers.CharField(
-        max_length=128,
-        min_length=8,
-        write_only=True
-    )
+    password = serializers.CharField(max_length=128, min_length=8, write_only=True)
 
     # The client should not be able to send a token along with a registration
     # request. Making `token` read-only handles that for us.
 
+    def reg_fields_validation(self, username, email, password):
+        # if email is None:
+        #     raise serializers.ValidationError("An email address is required")
+
+        if re.compile("[!@#$%^&*:;?><.0-9]").match(username):
+            raise serializers.ValidationError(
+                "Invalid Username , it contains invalid characters."
+            )
+
+        if str(password).isdigit() :
```

Comment:
use one regex for both alphanumeric e.g `^(?=.*[a-z])(?=.*\d)(?=.*[A-Z])([a-zA-Z\d]{8})$` pattern can take care of your password validations instead of multiple if statements.

Reply:
Done and it turns out it's very effective.